### PR TITLE
Remove nested designs data

### PR DESCRIPTION
**Pull request by 2potatocakes**
This PR addresses the discussion had earlier today between the Sites and Hosted teams regarding the `design` models file structure. It was agreed upon to try and abide by the following rules.

- 1 design per 1 .json file
- design data within the file will be stored as a JSON Object and **not** within a JSON Array
- design json files will **not** be nested in directories any deeper than the `data/designs/[vendor]` folder
- each file will be named either `[slug]-[uuid].json` or `[uuid].json`

I also refactored the `Designs::DataSource` model to only look for data within specified data directories, namely `data/designs/[vendor]` where `vendor` is defined within our hardcoded list of `WebsitesCatalogue::VENDORS`

### CODE REVIEW

File: lib/catalogue/designs/data_source.rb [hunk 1]
```diff
@@ -6,26 +6,25 @@ def self.read
       end
 
       def read
-        designs_paths.flat_map { |path| processed_json(path) }
+        designs_paths.map { |vendor, path| processed_json(vendor, path) }
```
Comment (smlparry) [inline lib/catalogue/designs/data_source.rb]: Where does `vendor` come from 🤔 ?

Comment (2potatocakes) [inline lib/catalogue/designs/data_source.rb]: I've renamed `designs_paths` ~> `vendor_design_paths` and updated the variable names, does that help clear it up? The `vendor_design_paths` method returns an array of arrays that look like this:
`[[vendor, path],[vendor, path]]`. When this is `map`ped in the bit you highlighted above, each of the args becomes available to the block. Does that make sense?

Comment (smlparry) [inline lib/catalogue/designs/data_source.rb]: Ah right, yeah that makes sense 👍

File: lib/catalogue/designs/data_source.rb [hunk 2]
```diff
@@ -6,26 +6,25 @@ def self.read
...
-      def designs_paths
-        Dir.glob(WebsitesCatalogue.config.designs_data.join('**/*.json'))
-      end
-
-      def vendor_name_from_path(path)
-        path.match(/^#{WebsitesCatalogue.config.designs_data}\/([^\/]*)\//i)[1].tap do |vendor|
-          raise "Unrecognised vendor: #{vendor}" unless WebsitesCatalogue::VENDORS.include?(vendor)
+      def vendor_design_paths
+        WebsitesCatalogue::VENDORS.inject([]) do |vendor_paths, vendor|
+          Dir.glob(WebsitesCatalogue.config.designs_data.join(vendor, "*.json")).each do |path|
```
Comment (smlparry) [inline lib/catalogue/designs/data_source.rb]: Could this be a `map` and therefore not have to return the variable?

Comment (2potatocakes) [inline lib/catalogue/designs/data_source.rb]: Do you mean in place of the `inject` method or in addition to it? Sorry I'm not sure what you mean? Feel free to paste up code block if you think this code can be tidied up a bit more

Comment (travo) [inline lib/catalogue/designs/data_source.rb]: Having done a couple of refactoring experiments against @2potatocakes suite of excellent tests, I think the code he's written here is tight.

The only thing I'd suggest would look something like this;

```ruby
      def vendor_data_files(vendor)
        WebsitesCatalogue.config.designs_data.join(vendor, "*.json")
      end

      def vendor_design_paths
        WebsitesCatalogue::VENDORS.inject([]) do |vendor_paths, vendor|
          Dir.glob(vendor_data_files(vendor)).map do |path|
            vendor_paths << [vendor, path]
          end
          vendor_paths
        end
      end
```

**Pull request merged**